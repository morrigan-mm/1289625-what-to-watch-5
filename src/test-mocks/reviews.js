const reviews = [
  {
    id: 1,
    text: `some review text 1`,
    rating: 1.1,
    author: `Author name1`,
    reviewDate: `2020-01-01T14:13:56.569Z`
  },
  {
    id: 2,
    text: `some review text 2`,
    rating: 2.2,
    author: `Author name2`,
    reviewDate: `2020-02-02T14:13:56.569Z`
  },
  {
    id: 3,
    text: `some review text 3`,
    rating: 3.3,
    author: `Author name3`,
    reviewDate: `2020-03-03T14:13:56.569Z`
  }
];

export default reviews;
