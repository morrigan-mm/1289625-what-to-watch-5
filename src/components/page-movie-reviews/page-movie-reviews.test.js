import React from "react";
import renderer from "react-test-renderer";
import PageMovieReviews from "./page-movie-reviews";
import reviews from "../../test-mocks/reviews";

const emptyReviews = [];

describe(`PageMovieReviews`, () => {
  it(`Should render correctly with full review list`, () => {
    const tree = renderer
      .create(
          <PageMovieReviews reviews={reviews} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render correctly with empty review list`, () => {
    const tree = renderer
      .create(
          <PageMovieReviews reviews={emptyReviews} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


