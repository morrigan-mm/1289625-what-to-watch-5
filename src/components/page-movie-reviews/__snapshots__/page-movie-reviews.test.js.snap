// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMovieReviews Should render correctly with empty review list 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <p
    className="review"
  >
    There are currently no reviews for this movie
  </p>
</div>
`;

exports[`PageMovieReviews Should render correctly with full review list 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          some review text 1
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Author name1
          </cite>
          <time
            className="review__date"
            dateTime="1/1/2020"
          >
            January 1, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        1.1
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          some review text 2
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Author name2
          </cite>
          <time
            className="review__date"
            dateTime="2/2/2020"
          >
            February 2, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        2.2
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          some review text 3
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Author name3
          </cite>
          <time
            className="review__date"
            dateTime="3/3/2020"
          >
            March 3, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.3
      </div>
    </div>
  </div>
</div>
`;
